crypto/http/http_client.c

```diff
@@ -0,0 +1,1204 @@
+            BIO_printf(fbio, "Proxy-Authorization: Basic %s\r\n", proxyauthenc);
+            OPENSSL_clear_free(proxyauthenc, strlen(proxyauthenc));
+        }
+     proxy_end:
+        OPENSSL_clear_free(proxyauth, len);
+        if (proxyauthenc == NULL)
+            goto end;
+    }
+
+    /* Terminate the HTTP CONNECT request */
+    BIO_printf(fbio, "\r\n");
+
+    for (;;) {
+        if (BIO_flush(fbio) != 0)
+            break;
+        /* potentially needs to be retried if BIO is non-blocking */
+        if (!BIO_should_retry(fbio))
+            break;
+    }
+
+    for (;;) {
+        /* will not actually wait if timeout == 0 */
+        rv = BIO_wait(fbio, max_time);
+        if (rv <= 0) {
+            BIO_printf(bio_err, "%s: HTTP CONNECT %s\n", prog,
+                       rv == 0 ? "timed out" : "failed waiting for data");
+            goto end;
+        }
+
+        /*-
+         * The first line is the HTTP response.
+         * According to RFC 7230, it is formatted exactly like this:
+         * HTTP/d.d ddd Reason text\r\n
+         */
+        read_len = BIO_gets(fbio, mbuf, BUF_SIZE);
+        /* the BIO may not block, so we must wait for the 1st line to come in */
+        if (read_len < HTTP_LINE1_MINLEN)
+            continue;
+
+        /* RFC 7231 4.3.6: any 2xx status code is valid */
+        if (strncmp(mbuf, HTTP_PREFIX, strlen(HTTP_PREFIX)) != 0) {
+            HTTPerr(0, HTTP_R_SERVER_RESPONSE_PARSE_ERROR);
+            BIO_printf(bio_err, "%s: HTTP CONNECT failed, non-HTTP response\n",
+                       prog);
+            /* Wrong protocol, not even HTTP, so stop reading headers */
+            goto end;
+        }
+        mbufp = mbuf + strlen(HTTP_PREFIX);
+        if (strncmp(mbufp, HTTP_VERSION_PATT, strlen(HTTP_VERSION_PATT)) != 0) {
+            HTTPerr(0, HTTP_R_SERVER_SENT_WRONG_HTTP_VERSION);
+            BIO_printf(bio_err,
+                       "%s: HTTP CONNECT failed, bad HTTP version %.*s\n",
+                       prog, HTTP_VERSION_STR_LEN, mbufp);
+            goto end;
+        }
+        mbufp += HTTP_VERSION_STR_LEN;
+        if (strncmp(mbufp, " 2", strlen(" 2")) != 0) {
+            mbufp += 1;
+            /* chop any trailing whitespace */
+            while (read_len > 0 && ossl_isspace(mbuf[read_len - 1]))
+                read_len--;
+            mbuf[read_len] = '\0';
+            HTTPerr(0, HTTP_R_CONNECT_FAILURE);
+            ERR_add_error_data(2, "Reason=", mbufp);
+            BIO_printf(bio_err, "%s: HTTP CONNECT failed, Reason=%s\n",
+                       prog, mbufp);
+            goto end;
+        }
+        ret = 1;
+        break;
+    }
+
+    /* Read past all following headers */
+    do {
+        /*
+         * TODO: This does not necessarily catch the case when the full
+         * HTTP response came in in more than a single TCP message.
+         */
+        read_len = BIO_gets(fbio, mbuf, BUF_SIZE);
+    }
+    while (read_len > 2);
```

Comment:
Nit: while should be on the same line as }

Reply:
Fixed, as well as two similar code formatting nits in `http_lib.c`.
I've also added checks for situations like this in the format checking tool #10363.